lighthouse-core/scripts/lantern/collect/collect.js

```diff
@@ -226,59 +231,70 @@ async function main() {
     // Can run in parallel.
     const wptResultsPromises = [];
     for (let i = 0; i < SAMPLES; i++) {
-      const resultPromise = repeatUntilPass(() => runForWpt(url));
+      const resultPromise = repeatUntilPassOrEmpty(() => runForWpt(url));
       // Push to results array as they finish, so the progress indicator can track progress.
       resultPromise.then((result) => wptResults.push(result)).finally(updateProgress);
       wptResultsPromises.push(resultPromise);
     }
 
-    // Wait for WPT to finish, because it can take quite awhile to start and we want
-    // to avoid seeing totally different content locally.
-    await Promise.all(wptResultsPromises);
+    // Wait for the first WPT result to finish because we can sit in the queue for a while before we start
+    // and we want to avoid seeing totally different content locally.
+    await Promise.race(wptResultsPromises);
 
     // Must run in series.
     for (let i = 0; i < SAMPLES; i++) {
-      const resultPromise = repeatUntilPass(() => runUnthrottledLocally(url));
+      const resultPromise = repeatUntilPassOrEmpty(() => runUnthrottledLocally(url));
       unthrottledResults.push(await resultPromise);
       updateProgress();
     }
 
+    // Wait for *all* WPT runs to finish since we just waited on the first one earlier.
+    await Promise.all(wptResultsPromises);
+
     const urlResultSet = {
       url,
-      wpt: wptResults.map((result, i) => {
+      wpt: wptResults.filter(result => result.lhr && result.trace).map((result, i) => {
+        if (!result.lhr || !result.trace) throw new Error('Expected lhr and trace');
```

Comment:
can remove error since there is a filter?

Reply:
it's mostly for ts, comment for ya ok?